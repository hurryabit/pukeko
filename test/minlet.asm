g_declare_cafs main
g_declare_main main

g_globstart print, 2
g_eval
g_print
g_cons 0, 0
g_cons 0, 2
g_update 1
g_return

g_globstart id, 1
g_push 0
g_push 0
g_eval
g_update 3
g_pop 2
g_unwind

g_globstart main, 0
g_pushint 0
g_pushglobal id, 1
g_mkap 1
g_pushglobal print, 2
g_mkap 1
g_update 1
g_pop 0
g_unwind
