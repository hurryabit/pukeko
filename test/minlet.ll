external print : Int -> IO Unit = "print"
let id$ll1 : ∀a. a -> a =
      fun @a ->
        fun (x : a) ->
          let y : a = x in
          y
let id : ∀a. a -> a = id$ll1
let main : IO Unit = print (id @Int 0)
