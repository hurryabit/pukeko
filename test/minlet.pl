type Unit =
       | Unit
type Pair a b =
       | Pair a b
type Bool =
       | False
       | True
type Int
type List a =
       | Nil
       | Cons a (List a)
type IO a
external print : Int -> IO Unit = "print"
let id : âˆ€a. a -> a =
      fun @a ->
        fun (x : a) ->
          let y : a = x in
          y
let main : IO Unit = print (id @Int 0)
