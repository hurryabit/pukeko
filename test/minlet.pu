let id x = let y = x in y

let main = print (id 0)
