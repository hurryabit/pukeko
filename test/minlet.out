
Steps  :         95
Allocs :          3          (Checks:          3)
GC runs:          0
